src/vm/eventpipe.cpp

```diff
@@ -460,19 +490,26 @@ void EventPipe::DisableInternal(EventPipeSessionID id, EventPipeProviderCallback
 EventPipeSession *EventPipe::GetSession(EventPipeSessionID id)
 {
     LIMITED_METHOD_CONTRACT;
-    _ASSERTE(IsLockOwnedByCurrentThread());
```

Comment:
We need some form of synchronization to ensure that the EventPipeSession* that we are returning here won't be deleted. Previously IsLockOwnedByCurrentThread() was providing that synchronization. Either that needs to be restored, some other form of synchronization needs to be added, or some larger refactoring needs to occur that eliminates this method entirely.